### PR TITLE
implement folder creation for data saving

### PR DESCRIPTION
**Pull request by Lherrera094**

### CODE REVIEW

File: src/save_data.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#include "save_data.h"
+
+void create_folder(saveData *saveDCfg){
+
+    simulation_folder( projectPath );
+    data_folder( projectPath, foldername );
+    copyJSON( projectPath, foldername );
+
+    exit(-1);
```
Comment (alfkoehn) [inline src/save_data.c]: I think this is left from debugging

Comment (Lherrera094) [reply src/save_data.c]: Oh my bad, never happening again

File: src/save_data.c [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        }else{
+            printf("Error creating directory: %s\n", path);
+            return;
+        }
+    }
+}
+
+void data_folder(const char *path, const char *folder_name){
+    
+    char fullPath[1024];
```
Comment (alfkoehn) [inline src/save_data.c]: we might want to use PATH_MAX instead

File: src/save_data.c [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            return;
+        }
+    }else{
+        printf("%s already exists.\n", folder_name);
+    }
+}
+
+void copyJSON(const char *path, const char *folder_name){
+    
+    char destination[1024];
```
Comment (alfkoehn) [inline src/save_data.c]: again, PATH_MAX

**Review by alfkoehn — CHANGES_REQUESTED** [review]
remove the exit(-1) in save_data.c

**Comment by Lherrera094** [comment]
Add PATH_MAX

File: src/save_data.c [hunk 4]
```diff
@@ -6,8 +6,6 @@ void create_folder(saveData *saveDCfg){
     data_folder( projectPath, foldername );
     copyJSON( projectPath, foldername );
 
-    exit(-1);
-
```
Comment (alfkoehn) [inline src/save_data.c]: :-)

**Review by alfkoehn — APPROVED** [review]
thanks for adjusting

**Pull request merged**